spec/system/views/posts/index_spec.rb

```diff
@@ -0,0 +1,93 @@
+require 'rails_helper'
+
+RSpec.describe 'Post#Index', type: :system do
+  describe 'The Post#Index page' do
+    it 'displays the users profile picture' do
+      User.all.each do |user|
+        visit user_posts_path(user.id)
+        expect(page).to have_css("img[src*='#{user.photo}']")
+      end
+    end
+
+    it 'displays the username' do
+      User.all.each do |user|
+        visit user_posts_path(user.id)
+        expect(page).to have_content(user.name)
+      end
+    end
+
+    it 'displays the number of posts' do
+      User.all.each do |user|
+        visit user_posts_path(user.id)
+        expect(page).to have_content(user.posts.count)
+      end
+    end
+
+    it 'displays the posts title' do
+      User.all.each do |user|
+        visit user_posts_path(user.id)
+        user.posts.take(3).each do |post|
+          expect(page).to have_content(post.title)
+        end
+      end
+    end
+
+    it 'displays the See more button' do
+      User.all.each do |user|
+        visit user_posts_path(user.id)
+        if user.posts.count > 3
+          expect(page).to have_content('Show more posts')
+        else
+          expect(page).to_not have_content('Show more posts')
+        end
+      end
+    end
+
+    it 'displays the posts text' do
+      User.all.each do |user|
+        visit user_posts_path(user.id)
+        user.posts.take(3).each do |post|
+          expect(page).to have_content(post.text)
+        end
+      end
+    end
+
+    it 'displays the first 3 comments of the first post' do
+      User.all.each do |user|
+        visit user_posts_path(user.id)
+        user.posts.first.comments.take(3).each do |comment|
+          expect(page).to have_content(comment.text)
+        end
+      end
+    end
+
+    it 'displays the comments counter' do
+      User.all.each do |user|
+        visit user_posts_path(user.id)
+        user.posts.take(3).each do |post|
+          expect(page).to have_content(post.comments_counter)
+        end
+      end
+    end
+
+    it 'display the likes counter' do
+      User.all.each do |user|
+        visit user_posts_path(user.id)
+        user.posts.each do |post|
+          expect(page).to have_content(post.likes_counter)
+        end
+      end
+    end
+
+    it 'redirects to the post when clicked on' do
+      User.all.each do |user|
+        visit user_posts_path(user.id)
+        user.posts.take(3).each do |post|
+          click_on post.title
+          expect(page).to have_current_path(user_post_path(user.id, post.id))
+          visit user_posts_path(user.id)
+        end
+      end
+    end
```

Comment:
- [ ] As mentioned by the previous reviewer and for the project requirements the `User` **post index page** has **tests that asserts you can see a section for pagination if there are more posts than fit on the view.**

Reply:
My tests are passing:

![Passing_tests](https://github.com/limavfabio/MyRailsBlog/assets/92121463/8b999289-545c-4238-a012-87143b6c67a5)